mainapp/src/Admin.tsx

```diff
@@ -25,13 +25,37 @@ const style = {
 
 };
 
+type EditionTab
+    = 'UsersEdition' | 'ProfilesEdition' | 'SourcesEdition'
+
+const editionTabToNumber = (editionTab: EditionTab) => {
+    switch (editionTab) {
+        case 'UsersEdition': return 0
+        case 'ProfilesEdition': return 1
+        case 'SourcesEdition': return 2
+        default: 0
+    }
+}
+
+const editionTabToString = (editionTab: number): EditionTab => {
+    switch (editionTab) {
+        case 0: return 'UsersEdition'
+        case 1: return 'ProfilesEdition'
+        case 2: return 'SourcesEdition'
+        default: return 'UsersEdition'
+    }
+}
+
+
 type Model = {
     users: RD<string, User[]>,
     profiles: RD<string, string[]>,
-    isModalOpen: boolean
+    isModalOpen: boolean,
```

Comment:
one solution is to let a trailing comma to avoid modify  2 lines when only once is added.